todo/src/components/TodoList/model/TodoListManager.js

```diff
@@ -92,4 +92,38 @@ export class TodoListManager {
   nextState() {
     this.#filterState.next();
   }
+
+  moveItemIndex(from, to) {
+    switch (this.#filterState.currentState) {
+      case CONSTANTS.CHECKED_STATE:
+        this.moveItem(this.#checkedList, from, to);
+        break;
+      case CONSTANTS.UNCHECKED_STATE:
+        this.moveItem(this.#uncheckedList, from, to);
+        break;
+      case CONSTANTS.ALL_STATE:
+        if (this.todoList[from].checked) {
+          from -= this.#uncheckedList.length;
+          to -= this.#uncheckedList.length;
+          this.moveItem(this.#checkedList, from, to);
+        } else {
+          to = to > this.#uncheckedList.length ? this.#uncheckedList.length : to;
```

Comment:
```suggestion
          to = Math.min(this.#uncheckedList.length, to);
```
`min` 함수 사용해도 되겠네요~!

Reply:
수정했습니다!